### PR TITLE
VideoPress dashboard: improve upload error handler

### PR DESCRIPTION
**Pull request by CGastrell**
Fixes https://github.com/Automattic/videopress/issues/1038

## Proposed changes:
This PR addresses unhandled upload errors on VideoPress dashboard, providing error components for both list and row views of VideoPress dashboard.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

The next instructions should be tested on:
- [ ] self hosted/JN
- [ ] simple sites (simple sites don't use VideoPress dashboard, but uploads should be tested on these as well)
- [ ] AT sites (should mostly behave as self hosted, since the standalone plugin takes place, but unsure how to get the latest code builds to AT)

Create an instance, follow the upgrade flow for VideoPress. Once done, you should be able to access VideoPress' dashboard at `wp-admin/admin.php?page=jetpack-videopress#/`

- Verify the video uploads work on dashboard
- Verify the video uploads work on the block editor
- Verify the video uploads work on the media library

The upload process should not show any improvements or handling, but it should work as before.

- Install "Disable XML-RPC" [plugin](https://wordpress.org/plugins/disable-xml-rpc-api/) and activate it

At this point, the XML backend connection should not be working, hence, video uploads should fail

Go back to VideoPress dashboard and upload a video. It should fail and the video card should show a warning sign. Click on the button to get some context and a link to our troubleshooting guide. Click on the trash icon to get rid of the error.

image
image

Do it again, but switch to list/grid view and verify the same functionality exists.
image
image

Disabling "Disable XML-RPC" plugin should get things back to normal, but beware of this step on local environments as it turns to get tricky (unsure why).

### CODE REVIEW

File: projects/packages/videopress/src/client/lib/resumable-file-uploader/index.ts [hunk 1]
```diff
@@ -54,22 +57,35 @@ const resumableFileUploader = ( {
...
 	const upload = new tus.Upload( file, {
-		onError: onError,
+		onError,
 		onProgress,
 		endpoint: tokenData.url,
 		removeFingerprintOnSuccess: true,
-		withCredentials: false,
-		autoRetry: true,
+		// withCredentials: false,
+		// autoRetry: true,
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/lib/resumable-file-uploader/index.ts]: Not needed?

Comment (CGastrell) [reply projects/packages/videopress/src/client/lib/resumable-file-uploader/index.ts]: Actually no use for those. There are no such options on the underlying lib

**Review by dhasilva — APPROVED** [review]
LGTM. Very well done! Thanks!

**Comment by foleynotrose** [comment]
Thanks @CGastrell ! Two questions: 

1. Do you know when will this be released? 
2. What help doc does the `Visit the docs` link to?

**Comment by CGastrell** [comment]
> Thanks @CGastrell ! Two questions:
> 
> 1. Do you know when will this be released?
> 2. What help doc does the `Visit the docs` link to?

Releasing today if everything goes well
Currently links to https://jetpack.com/support/jetpack-videopress/troubleshooting-video-hosting-issues/

**Comment by foleynotrose** [comment]
Thanks! I've updated that article to make it clear that this problem could be due to a broken Jetpack Connection.

**Pull request merged**